### DISCUSSION TITLE
One to Many relations

### DISCUSSION THREAD

**Discussion by ale1981** [Laravel-Backpack/community-forum#723] [Q&A (Help)] [answered]
I'm struggling to get the URLs for a one to many relation to work as it would within a normal Laravel deployment.

I have a customer model which has a one to many relation to the address model.

 I have attempted to amend the CRUD::setRoute() but it's not working, ideally the following would be possible?

list - /admin/customer/{customer_id}/addresses
edit - /admin/customer/{customer_id}/address/{id}

**Comment by phpfour** [comment]
Hey @ale1981, thanks for posting here. 

Can you please add some of your code here and point us to what you are trying to achieve? At this moment, it's not clear what exact issue you are facing.

Thanks!

**Comment by ale1981** [comment]
Hi @phpfour 

Not sure code is going to help but I will try and explain a bit better.

I have two models, a customer model and a customer address model.
A customer can have many addresses.
A customer address belongs to a customer.

I have created a CustomerCrudController and a CustomerAddressCrudController.

I only want the CustomerAddressCrudController to be accessed **through** a customer.
Currently I can list ALL customer addresses by going to /admin/customer-addresses, however there is no point to this as an address can't exist without a customer.

What I want is to view a particular customers' addresses by visiting /admin/customer/{customer_id}/customer-addresses.
In this example the show and update operations would point to /admin/customer/{customer_id}/customer-addresses/{customer_address_id}

I can list the relation on the CustomerCrud but when you have lots of results the relation column is no good.

I hope that it's a little clearer?!

**Comment by ale1981** [comment]
Nobody else had this problem with one to many relations?

**Reply by pxpm** [reply, marked as answer]
Hey @ale1981 sorry for the late reply. 

It seems this issue fell through the cracks 👎 

From your description, it seems you are trying to create a nested crud ? 

If so, you have an example of that in our demo: https://github.com/Laravel-Backpack/demo/blob/main/app/Http/Controllers/Admin/PetShop/OwnerPetsCrudController.php

Let me know if that helps. 

Cheers

**Comment by ale1981** [comment]
Hi @pxpm 

Thanks, this has definitely put me in the right direction.

I will let you know if it works as expected.


Thanks
